backend/inspirehep/refextract/utils.py

```diff
@@ -0,0 +1,86 @@
+import redis
+import structlog
+from flask import current_app
+from invenio_records.models import RecordMetadata
+from sqlalchemy import cast, not_, type_coerce
+from sqlalchemy.dialects.postgresql import JSONB
+
+from inspirehep.matcher.utils import normalize_title
+
+LOGGER = structlog.getLogger()
+JOURNAL_DICT_REDIS_EXPIRATION_PERIOD = 60 * 60
+
+
+def create_journal_dict():
+    """
+    Returns a dictionary that is populated with refextracts's journal KB from the database.
+
+        { SOURCE: DESTINATION }
+
+    which represents that ``SOURCE`` is translated to ``DESTINATION`` when found.
+
+    Note that refextract expects ``SOURCE`` to be normalized, which means removing
+    all non alphanumeric characters, collapsing all contiguous whitespace to one
+    space and uppercasing the resulting string.
+    """
+    only_journals = type_coerce(RecordMetadata.json, JSONB)["_collections"].contains(
+        ["Journals"]
+    )
+    only_not_deleted = not_(
+        type_coerce(RecordMetadata.json, JSONB).has_key("deleted")  # noqa
+    ) | not_(  # noqa
+        type_coerce(RecordMetadata.json, JSONB)["deleted"] == cast(True, JSONB)
+    )
+    entity_short_title = RecordMetadata.json["short_title"]
+    entity_journal_title = RecordMetadata.json["journal_title"]["title"]
+    entity_title_variants = RecordMetadata.json["title_variants"]
+
+    titles_query = RecordMetadata.query.with_entities(
+        entity_short_title, entity_journal_title
+    ).filter(only_journals, only_not_deleted)
+
+    title_variants_query = RecordMetadata.query.with_entities(
+        entity_short_title, entity_title_variants
+    ).filter(only_journals, only_not_deleted)
+
+    title_dict = {}
+
+    for (short_title, journal_title) in titles_query.all():
+        title_dict[normalize_title(short_title)] = short_title
+        title_dict[normalize_title(journal_title)] = short_title
+
+    for (short_title, title_variants) in title_variants_query.all():
+        if title_variants is None:
+            continue
+
+        sub_dict = {
+            normalize_title(title_variant): short_title
+            for title_variant in title_variants
+        }
+
+        title_dict.update(sub_dict)
+
+    return title_dict
+
+
+def _write_journal_kb_dict_to_redis(redis):
+    journal_dict = create_journal_dict()
+    LOGGER.info("Writing journal KB dict to redis")
+    redis.hmset("refextract_journal_kb", journal_dict)
+    redis.expire("refextract_journal_kb", JOURNAL_DICT_REDIS_EXPIRATION_PERIOD)
+
+    return journal_dict
```

Comment:
do you really need to store this in redis for performance? I'm a bit concerned about data consistency.

Reply:
On next we have a celery beat which is downloading it to the file every hour. With @drjova we discussed that we won't use that anymore and we'll store data in redis and recalculate with >= 1h

Reply:
because the old `refextract` version that `next` is using needs to have the file on disk, so we need to generate it periodically. The new one on `hep` doesn't, and the DB query to generate the data is fast, so I don't believe it's worth it to reintroduce potential inconsistencies and complexity for very little performance gain:
```
In [1]: from inspirehep.matcher.utils import create_journal_dict                                                                                                

In [2]: %timeit create_journal_dict()                                                                                                                           
109 ms ± 3.81 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

Reply:
we have already timed it, the point is if we have 1k papers coming from arxiv every morning we don't want to hammer db just for that.

Reply:
ok, can we decrease the expiration to something like a few secs - 1 min max then and/or invalidate the cache on `JournalRecord` CRUD?

Reply:
Yep, that's a good idea to expire it there

Reply:
mmmh, now I think of it it needs to be done in the DB commit receiver to avoid the race condition where the query happens between calling the CRUD method and the DB commit.

Reply:
How often do we update/add journals?

Reply:
rarely, less than once a day I would say.

Reply:
So we are talking about a use case that it might or not happen once a day. Then I agree, we can add it to the receiver and avoid the unnecessary queries.